catalogue_api/api/src/test/scala/uk/ac/wellcome/platform/api/responses/ResultListResponseTest.scala

```diff
@@ -0,0 +1,110 @@
+package uk.ac.wellcome.platform.api.responses
+
+import com.twitter.finagle.http.{Method, Request}
+import org.scalatest.{FunSpec, Matchers}
+import uk.ac.wellcome.display.models.v1.DisplayWorkV1
+import uk.ac.wellcome.platform.api.models.DisplayResultList
+import uk.ac.wellcome.platform.api.requests.MultipleResultsRequest
+
+class ResultListResponseTest extends FunSpec with Matchers {
+  val contextUri = "https://example.org/context.json"
+
+  val requestBaseUri = "https://api.example.org"
+  val requestUri = "/works"
+
+  val displayResultList = DisplayResultList[DisplayWorkV1](
+    pageSize = 10,
+    totalPages = 5,
+    totalResults = 45,
+    // Nothing checks that results is populated correctly!
+    results = List()
+  )
+
+  val multipleResultsRequest = MultipleResultsRequest(
+    page = 1,
+    pageSize = Some(displayResultList.pageSize),
+    includes = None,
+    id = None,
+    query = None,
+    _index = None,
+    request = Request(method = Method.Get, uri = requestUri)
+  )
+
+  describe("nextPage") {
+    it("omits the parameter if this is the only page") {
+      val resp = getResponse(
+        displayResultList = displayResultList.copy(totalPages = 1)
+      )
+
+      resp.nextPage shouldBe None
+    }
+
+    it("omits the parameter if this is the last page") {
+      val resp = getResponse(
+        displayResultList = displayResultList.copy(totalPages = 5),
+        multipleResultsRequest = multipleResultsRequest.copy(page = 5)
+      )
+
+      resp.nextPage shouldBe None
+    }
+
+    it("omits the parameter if the request is beyond the last page") {
+      val resp = getResponse(
+        displayResultList = displayResultList.copy(totalPages = 5),
+        multipleResultsRequest = multipleResultsRequest.copy(page = 10)
+      )
+
+      resp.nextPage shouldBe None
+    }
+
+    it("includes the parameter if there's a next page to view") {
+      val resp = getResponse(
+        displayResultList = displayResultList.copy(totalPages = 10),
+        multipleResultsRequest = multipleResultsRequest.copy(page = 5)
+      )
+
+      resp.nextPage shouldBe Some(s"$requestBaseUri$requestUri?page=6")
+    }
+  }
+
+  describe("prevPage") {
+    it("omits the parameter if this is the only page") {
+      val resp = getResponse(
+        displayResultList = displayResultList.copy(totalPages = 1)
+      )
+
+      resp.prevPage shouldBe None
+    }
+
+    it("omits the parameter if this is the first page") {
+      val resp = getResponse(
+        displayResultList = displayResultList.copy(totalPages = 5),
+        multipleResultsRequest = multipleResultsRequest.copy(page = 1)
+      )
+
+      resp.prevPage shouldBe None
+    }
+
```

Comment:
Maybe assert the next page in these cases as well? It's captured elsewhere in the tests, but might be good to cover.

Reply:
I’d rather keep them as-is – each test can only fail one way. The next/prev page logic is independent, so independent tests.

I’ll add a new test which tests than both at once, and keep this describe block tightly scoped.

Reply:
Yes, completely agree about tests having only a single reason to fail.  Single assertion is a good rule, but sometimes there's a balance with test duplication.